hybridse/src/udf/udf.cc

```diff
@@ -64,6 +64,45 @@ void hex(StringRef *str, StringRef *output) {
     output->data_ = buffer;
 }
 
+void unhex(StringRef *str, StringRef *output) {
+    std::ostringstream ss;
+    uint8_t* arr = new uint8_t[str->size_];
```

Comment:
since there will be one more function parameter: `is_null`, it's fine to alloc buffer at very begining, then just update the allocated string content in place.

Reply:
here is a simplified steps:
1. alloc the managed buffer with size `origin_size / 2 + origin_size % 2` and memset
2. iterate through the input and 
   - any invalid character will set `is_null` to true and break
   - update the pos of buffer with value `arr[i] << 4 | arr[i+1]`
3. write `output` if not null

Reply:
`buffer[i/2] = char(arr[i] << 4 | arr[i+1]);`, it works!
I thought there is no way to update the corresponde position of buffer but using `stringstream`, it's amazing, thanks for the kindly hint.:)

But there is one weird question: 
```shell
[ RUN      ] UdfIRBuilderTest.UnhexTest
/home/aucker/mldb/OpenMLDB/hybridse/src/udf/udf_test.h:172: Failure
Expected equality of these values:
  expect
    Which is: 7&�5
  output
    Which is: 7&�5
[  FAILED  ] UdfIRBuilderTest.UnhexTest (119 ms)
```
after I update the code, I test it again, all tests passed except this one, but in the log, they are all same, I can't see the difference between of `expect` and the `output`.
And the corresponde test is this one, I just remove the last `C`:
```shell
CheckUdf<StringRef, StringRef>("unhex", "7&�5", StringRef("537061726B2053514"));
```

Reply:
`7` is `0x55`, `%` is `0x37`

Reply:
`b2` is wrong

Reply:
Should I add another case to handle this error? I copy the unhex result from MySQL 5.7 as the `expect` result in test, so I guess when handle the hex result like `0xa5`, `0xd1` etc. Though each character is valid, but they are not in ASCII table, it will give the same error. Not so sure, I will have a try at 8am, my computer is not around.

Reply:
I tried in mysql and sparksql, for hex that are not in ascii table, it all returns the `�` character.
```shell
mysql> select unhex("80");
+-------------+
| unhex("80") |
+-------------+
| �           |
+-------------+
1 row in set (0.00 sec)

mysql> select unhex("88");
+-------------+
| unhex("88") |
+-------------+
| �           |
+-------------+
1 row in set (0.00 sec)
```
In sparksql: 
```shell
spark-sql> select unhex("80");
�
Time taken: 0.057 seconds, Fetched 1 row(s)
spark-sql> select unhex("81");
�
Time taken: 0.051 seconds, Fetched 1 row(s)
```

Github don't render the special characters, actually my test is 
![image](https://user-images.githubusercontent.com/41871212/191645585-c3e2f12c-60d1-493f-bfed-8a222f5ffca4.png), so just `B2` is not valid character, `B2` is like `80`, `81` etc. When hex value is greater than `7f`, it all return `�`.
I add two tests,
```shell
CheckUdf<StringRef, StringRef>("unhex", "�", StringRef("a2")); 
CheckUdf<StringRef, StringRef>("unhex", "�", StringRef("8a"));
```
and here is the log:
```shell
/home/aucker/mldb/OpenMLDB/hybridse/src/udf/udf_test.h:172: Failure
Expected equality of these values:
  expect
    Which is: �
  output
    Which is: �
/home/aucker/mldb/OpenMLDB/hybridse/src/udf/udf_test.h:172: Failure
Expected equality of these values:
  expect
    Which is: �
  output
    Which is: �
```

So I think now that this function behavior is like mysql and spark(return `�` when hex greater than `7F`),  and I cannot test which character is equal to the hex `�`, can I just remove this test?

Reply:
do not copy the character with your system clipboard. chars from 128 - 255 is what called the [extended ascii characters](https://www.ascii-code.com/). It has serval standards which result in different chars. You get a different char possibly because of the encoding setting of the terminal. 

So I think just write string result as `"\x80"` in tests is better. ?